src/cordial_sound/scripts/wav_file_publisher.py

```diff
@@ -18,35 +21,66 @@ def __init__(self):
             'cordial/sound/wav/header_length',
             24
         )
-        rospy.Subscriber(rospy.get_param('cordial_sound/play_wav_topic'), String, self.play_wav_file)
-        self._sound_publisher = rospy.Publisher(rospy.get_param('cordial_sound/play_stream_topic'), Sound, queue_size=1)
+        rospy.Subscriber(rospy.get_param(
+            'cordial_sound/play_wav_topic'), String, self.play_wav_file)
+        self._sound_publisher = rospy.Publisher(rospy.get_param(
+            'cordial_sound/play_stream_topic'), Sound, queue_size=10)
         self._pyaudio = pyaudio.PyAudio()
 
+        self.audio_chunks = []
+
+    def single_split_sec(self, from_sec, to_sec, audio):
+        t1 = from_sec * 1 * 1000
+        t2 = to_sec * 1 * 1000
+        split_audio = audio[t1:t2]
+        self.audio_chunks.append(split_audio)
+
+    def multiple_split_sec(self, filename, seconds_per_split):
+        audio = AudioSegment.from_wav(filename)
+        total_seconds = math.ceil(audio.duration_seconds)
+        for i in range(0, total_seconds, seconds_per_split):
+            single_split_sec(i, i+seconds_per_split, audio)
+            if i == total_seconds - seconds_per_split:
+                print('All splited successfully')
+
     def play_wav_file(self, data):
 
         file_path = data.data
 
         try:
-            wf = wave.open(file_path, 'rb')
+            wf_original = wave.open(file_path, 'rb')
         except IOError:
             rospy.logerr("Not a valid wav file: '{}'".format(file_path))
             return
 
-        audio_format = self._pyaudio.get_format_from_width(wf.getsampwidth())
-        framerate = wf.getframerate()
-        num_channels = wf.getnchannels()
+        # Splitting wav to multiple parts of 1 sec each
+        multiple_split_sec(file_path, seconds_per_split=1)
+
+        # Publishing each chunk of 1 sec separtely
+        for i in range(len(self.audio_chunks)):
+            self.audio_chunks[i].export('current_chunk.wav', format="wav")
+
+            wf = wave.open('current_chunk.wav', 'rb')
+
+            os.remove('current_chunk.wav')
```

Comment:
Could you just override the same file instead of creating and deleting the file? That's what we do with the general cordial audio: we always use the same temp file.